packages/create-llama/templates/components/ui/shadcn/chat/chat-messages.tsx

```diff
@@ -23,6 +24,12 @@ export default function ChatMessages(
   const showReload =
     props.reload && !props.isLoading && isLastMessageFromAssistant;
   const showStop = props.stop && props.isLoading;
+  const content = lastMessage?.content ?? ''
+  const deferredContent = useDeferredValue(content)
+  const isPending = props.isLoading ? !isLastMessageFromAssistant : (
+    props.isLoading &&
+    isLastMessageFromAssistant && content === deferredContent
+  )
```

Comment:
I have implemented it in another way.

I just got the idea from this section of the document (https://react.dev/reference/react/useDeferredValue#indicating-that-the-content-is-stale).

Reply:
I'll test this with the next release

Reply:
@himself65 You might also wanna have a look at https://github.com/marcusschiesser/ui - that's where the components are taken from